src/adapter/breakpoints/breakpointBase.ts

```diff
@@ -371,14 +371,23 @@ export abstract class Breakpoint {
     );
   }
 
-  private async _setByUrl(thread: Thread, url: string, lineColumn: LineColumn): Promise<void> {
+  private async _setByUrl(
+    thread: Thread,
+    url: string,
+    lineColumn: LineColumn,
+    filePath?: string,
+  ): Promise<void> {
     lineColumn = base1To0(lineColumn);
     if (this.hasSetOnLocation({ url }, lineColumn)) {
       return;
     }
 
+    const filePathRegexp = filePath
+      ? '|' + urlToRegex(absolutePathToFileUrl(filePath)?.toString()!)
+      : '';
+    const urlRegexp = urlToRegex(url) + filePathRegexp;
```

Comment:
`urlToRegex` already does this if a file uri is presented:

https://github.com/microsoft/vscode-js-debug/blob/7818330c650465c4e62b4d61cd9d7ed40101b284/src/common/urlUtils.ts#L321-L325